lib/class-wp-theme-json-gutenberg.php

```diff
@@ -590,27 +624,123 @@ private static function append_to_selector( $selector, $to_append ) {
-	 * the a preset slug and each value is the preset value.
+	 * Function that scopes a selector with another one. This works a bit like
+	 * SCSS nesting except the `&` operator isn't supported.
 	 *
-	 * @param array  $preset_per_origin Array of presets keyed by origin.
-	 * @param string $value_key         The property of the preset that contains its value.
+	 * <code>
+	 * $scope = '.a, .b .c';
+	 * $selector = '> .x, .y';
+	 * $merged = scope_selector( $scope, $selector );
+	 * // $merged is '.a > .x, .a .y, .b .c > .x, .b .c .y'
+	 * </code>
+	 *
+	 * @param string $scope    Selector to scope to.
+	 * @param string $selector Original selector.
+	 *
+	 * @return string Scoped selector.
+	 */
+	private static function scope_selector( $scope, $selector ) {
+		$scopes    = explode( ',', $scope );
+		$selectors = explode( ',', $selector );
+
+		$selectors_scoped = array();
+		foreach ( $scopes as $outer ) {
+			foreach ( $selectors as $inner ) {
+				$selectors_scoped[] = trim( $outer ) . ' ' . trim( $inner );
+			}
+		}
+
+		return implode( ', ', $selectors_scoped );
+	}
+
+	/**
+	 * Gets preset values keyed by slugs based on settings and metadata.
+	 *
+	 * <code>
+	 * $settings = array(
+	 *     'color' => array(
+	 *         'palette' => array(
+	 *             array(
+	 *                 'slug'       => 'sansSerif',
+	 *                 'fontFamily' => '"Helvetica Neue", sans-serif',
+	 *             ),
+	 *             array(
+	 *                 'slug'   => 'serif',
+	 *                 'colors' => 'Georgia, serif',
+	 *             )
+	 *         ),
+	 *     ),
+	 * );
+	 * $meta = array(
+	 *    'path'      => array( 'typography', 'fontFamilies' ),
+	 *    'value_key' => 'fontFamily',
+	 * );
+	 * $values_by_slug = get_settings_values_by_slug();
+	 * // $values_by_slug === array(
+	 * //   'sans-serif' => '"Helvetica Neue", sans-serif',
+	 * //   'serif'      => 'Georgia, serif',
+	 * // );
+	 * </code>
+	 *
+	 * @param array $settings Settings to process.
+	 * @param array $meta One of the PRESETS_METADATA values.
 	 *
 	 * @return array Array of presets where each key is a slug and each value is the preset value.
 	 */
-	private static function get_merged_preset_by_slug( $preset_per_origin, $value_key ) {
+	private static function get_settings_values_by_slug( $settings, $meta ) {
+		$preset_per_origin = _wp_array_get( $settings, $meta['path'], array() );
+
 		$result = array();
 		foreach ( self::VALID_ORIGINS as $origin ) {
 			if ( ! isset( $preset_per_origin[ $origin ] ) ) {
 				continue;
 			}
 			foreach ( $preset_per_origin[ $origin ] as $preset ) {
-				// We don't want to use kebabCase here,
-				// see https://github.com/WordPress/gutenberg/issues/32347
-				// However, we need to make sure the generated class or css variable
-				// doesn't contain spaces.
-				$result[ preg_replace( '/\s+/', '-', $preset['slug'] ) ] = $preset[ $value_key ];
+				$slug = gutenberg_experimental_to_kebab_case( $preset['slug'] );
```

Comment:
@oandregal Am I correct to assume that `gutenberg_experimental_to_kebab_case` is safe as a replacement?

Reply:
Read through the implementation in `trunk` and it seems it is, yeah. We transform the slugs to kebab case in all the places we use them, so doing it internally to that function is fine.

I've also prepared https://github.com/WordPress/gutenberg/pull/34787